doc/devel/coding_style_guideline.rst

```diff
@@ -0,0 +1,169 @@
+=============================
+DWI_ML Coding Style Guideline
+=============================
```

Comment:
We have already started adding such a page in the scil-documentation. Maybe you could improve this page: https://scil-documentation.readthedocs.io/en/latest/coding/scilpy.html
(you can create a PR in https://github.com/scilus/scil-documentation)

and here we could simply point there.

Actually, thinking about it, it would be nice if you did the same for your github explanation higher. We could just tell people that we ask for the same guidelines as the scil lab.

Reply:
Sorry, but I would just target this repository. I prefer to be cautious about `Scilpy`. Here we have the opportunity to decide on what we want our code and coding practices look like at this early stage.

Also, by the same token, one may argue why we do not adopt the `VITALabAI` style, in case the two encourage or adopt different styles.

Ideally we would be using `Scilpy` and `VITALabAI` as a `pip` installable packages, and the way things are documented there should not interfere they way things are documented here.

IMHO the `SCIL` documentation site is a necessary resource, and an extremely useful one for all lab members, but the coding practices should be written in a code repository's documentation resources, not in the documentation resources of a lab. Think that `SCIL` has other projects, or uses other scripting languages as well.

Reply:
The page I cited is not on the scilpy page but on the scil-documentation page. It means it could be general tools for all students in our lab. So I think it could still be moved to our doc.

I understand your point about scil vs vital. Do they already have an official coding style?

I just think that the work you are doing is very nice and exactly what I wanted to add in the scil-documentation. I would hate to just copy and paste your work there, and I would hate even more to have to remember to code differently when coding here or in scilpy.

Or at least we could see what's common. And then for the differences, we could deal with them. Ex, here we tend to add the annotation for parameters (ex: sft: StatefulTractogram = none) and not in scilpy. We could maybe explain it here in this smaller doc or then decide to do as in scilpy or then convince scilpy to do it.

Maybe others have an opinion? @mdesco

Reply:
> I understand your point about scil vs vital. Do they already have an official coding style?

There is no specific coding style guideline in either lab beyond an explicit mention to complying with `PEP8` in `VITALabAI`.

> I just think that the work you are doing is very nice and exactly what I wanted to add in the scil-documentation. I would hate to just copy and paste your work there, 

Thanks for this Emmanuel. I appreciate it a lot. As said, I am not willing to open this front at SCIL or VITAL.

> and I would hate even more to have to remember to code differently when coding here or in scilpy.

Different organizations may adopt different conventions. I personally find that if there is a tool that automatically formats the code, then I just do not need to worry too much about it. In my experience, if such a tool exists in a repository, then I do not care about the style as I would do otherwise, and this implies accepting a style that I may dislike if I had to format manually, and getting used to reading code with that style.

> Ex, here we tend to add the annotation for parameters (ex: sft: StatefulTractogram = none) and not in scilpy.

If those of you that have been using it find it useful, then 👍 for adding it to these guidelines and I would try my best to stick to it.,